IdentityCore/src/cache/crypto/MSIDAssymetricKeyPair.m

```diff
@@ -142,15 +146,24 @@ - (NSString *)kid
 {
     if (!_kid)
     {
-        NSString *jwk = [NSString stringWithFormat:s_jwkTemplate, self.keyExponent, self.keyModulus];
-        NSData *jwkData = [jwk dataUsingEncoding:NSUTF8StringEncoding];
+        NSData *jwkData = [self.publicKey dataUsingEncoding:NSUTF8StringEncoding];
         NSData *hashedData = [jwkData msidSHA256];
         _kid = [hashedData msidBase64UrlEncodedString];
     }
     
     return _kid;
 }
 
+- (NSString *)publicKey
+{
+    if (!_publicKey)
+    {
+        _publicKey = [NSString stringWithFormat:s_jwkTemplate, self.keyExponent, self.keyModulus];
```

Comment:
This is not what I'd expect to see when using publicKey property. Should this property be renamed?

Reply:
Addressed. Renamed it to stkJwk